internal/admission/eventsource/handler_test.go

```diff
@@ -0,0 +1,155 @@
+package eventsource_test
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"testing"
+
+	"github.com/kanopy-platform/argoslower/internal/admission/eventsource"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	admissionv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	common "github.com/argoproj/argo-events/pkg/apis/common"
+	esv1alpha1 "github.com/argoproj/argo-events/pkg/apis/eventsource/v1alpha1"
+)
+
+type FakeMeshChecker struct {
+	Mesh bool
+	Err  error
+}
+
+func (m *FakeMeshChecker) OnMesh(ns string) (bool, error) {
+	return m.Mesh, m.Err
+}
+
+func TestEventSourceHandler(t *testing.T) {
+
+	t.Parallel()
+	fmc := &FakeMeshChecker{
+		Mesh: true,
+	}
+
+	handler := eventsource.NewHandler(fmc)
+	scheme := runtime.NewScheme()
+	utilruntime.Must(esv1alpha1.AddToScheme(scheme))
+	decoder, err := admission.NewDecoder(scheme)
+	assert.NoError(t, err)
+	err = handler.InjectDecoder(decoder)
+	assert.NoError(t, err)
+
+	f := false
+	for _, test := range []struct {
```

Comment:
nit:   I think it is more readable if the anonymous struct is defined outside of the for...loop

Reply:
moved it out if you can approve again.